Simulator/conversions.py

```diff
@@ -88,61 +88,77 @@ def NED_to_ECI(vec_NED, glat, glon, GMST):
 #--------------------Quaternions-----------------------------
 
 def L(q):
+    """
+    Matrix equivilent to left-hand quaternion for mulitplication
+    (eg. q1 * q2 = L(q1)q2)
+    """
     s = q[0]
     v = q[1:4]
-    # I3 = np.eye(3)
-    # upper = np.hstack([s, -v])
-    # lower = np.column_stack([v, s*I3 + skew(v)])
-    # return  np.row_stack([upper, lower])
-    Vhat = skew(v)
     L = np.array([[s, -v[0], -v[1], -v[2]], 
-        [v[0], s, Vhat[0, 1], Vhat[0, 2]], 
-        [v[1], Vhat[1, 0], s, Vhat[1, 2]], 
-        [v[2], Vhat[2, 0], Vhat[2, 1], s]])
+            [v[0], s, -v[2], v[1]], 
+            [v[1], v[2], s, -v[0]], 
+            [v[2], -v[1], v[0], s]])
     return L
 
 
 def R(q):
+    """
+    Matrix equivilent to right-hand quaternion for mulitplication
+    (eg. q1 * q2 = R(q2)q1)
+    """
     s = q[0]
     v = q[1:4]
-    # I3 = np.eye(3)
-    # upper = np.hstack([s, -v])
-    # lower = np.column_stack([v, s*I3 - skew(v)])
-    # return  np.row_stack([upper, lower])
-    Vhat = skew(v)
     R = np.array([[s, -v[0], -v[1], -v[2]], 
-        [v[0], s, -Vhat[0, 1], -Vhat[0, 2]], 
-        [v[1], -Vhat[1, 0], s, -Vhat[1, 2]], 
-        [v[2], -Vhat[2, 0], -Vhat[2, 1], s]])
+        [v[0], s, v[2], -v[1]], 
+        [v[1], -v[2], s, v[0]], 
+        [v[2], v[1], -v[0], s]])
     return R
 
 
 def quat(v):
+    """
+    Normalizes a quaternion (or any 4d vector)
+    """
     assert len(v) == 4, f"can't make a quaternion out of: {v}"
     N = np.linalg.norm(v)
     assert N > 0, "Norm of input is 0. Can't make a quaternion out of this! Got {}".format(v)
     return v / N
 
-
 def conj(q):
+    """
+    Calculates the quaternion conjugate
+    """
     q = np.copy(q)
-    q[0] = -q[0]
+    q[1:4] = -q[1:4]
     return q
 
-
-def quatrot(quat, vec):
+def quatrot(q1, q2):
     """
     Rotates a vector using a quaternion.
     """
-    x = np.append(0, vec)
-    rotated = L(quat) @ R(quat).T @ x
-    return rotated[1:4]
+    vector = False
+    if len(q2) == 3:
+        q2 = np.append(0, q2)
+        vector = True
+
+    rotated = L(q1) @ R(q1).T @ q2
+
+    if vector:
+        return rotated[1:4]
+    else:
+        return rotated
 
 def quatmult(q1, q2):
     """
     Multiplies two quaternions.
     """
-    return L(q1) @ q2
+    s = q1[0]
+    v = q1[1:4]
+    L = np.array([[s, -v[0], -v[1], -v[2]], 
+            [v[0], s, -v[2], v[1]], 
+            [v[1], v[2], s, -v[0]], 
+            [v[2], -v[1], v[0], s]])
+    return L @ q2
```

Comment:
I don't think there's any reason here not to use the `L()` function like it was before. Especially now that you've removed the unnecessary allocation from L/R